### PR TITLE
BOAC-2040, s3 scheme for BOA-sourced attachments; consolidate logic under note.create|update

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2040

* I now remove the proof-of-concept `'/api/notes/attachment/upload'`
* the attachment upload/delete logic is under the hood in Note.create|update

### CODE REVIEW

File: boac/models/note.py [hunk 1]
```diff
@@ -134,3 +162,26 @@ def to_api_json(self):
             'createdAt': self.created_at,
             'updatedAt': self.updated_at,
         }
+
+
+def _create_attachment(note, name, byte_stream, uploaded_by):
+    # note = Note.find_by_id(note_id=note_id)
```
Comment (raydavis) [inline boac/models/note.py]: Stray?

Comment (johncrossman) [reply boac/models/note.py]: Yup. Fixin'

File: boac/models/note.py [hunk 2]
```diff
@@ -95,11 +108,25 @@ def refresh_search_index(cls):
         std_commit()
 
     @classmethod
-    def update(cls, note_id, subject, body):
+    def update(cls, note_id, subject, body, attachments=(), delete_attachment_ids=()):
```
Comment (raydavis) [inline boac/models/note.py]: Just curious -- why an empty tuple instead of an empty list?

File: boac/models/development_db.py [hunk 3]
```diff
@@ -255,40 +252,6 @@ def create_cohorts():
...
-            Rock 'n Roll rang sweet as victory, under neon signs
-        """,
-    )
-    base_dir = app.config['BASE_DIR']
-    NoteAttachment.create(
-        note_id=note.id,
-        path_to_attachment=f'{base_dir}/fixtures/mock_advising_note_attachment_1.txt',
-        uploaded_by_uid='6446',
-    )
-    std_commit(allow_test_environment=True)
```
Comment (johncrossman) [inline boac/models/development_db.py]: The above was moved to `pytest.fixture` in conftest

**Pull request merged**